Fix #9996 and #15622

janwiebe-jump opened the pull request:
As discussed in #9996 the CPListSection constructors are not fully correct, since the CPListImageRowItem can't be added to it, which is possible in the native iOS SDK.
This PR fixes that. With that, also #15622 seems to be fixed.

mandel-macaque commented:
/azp run

mandel-macaque commented on src/carplay.cs:
```diff
@@ -573,10 +573,10 @@ interface CPListItem : CPSelectableListItem, NSSecureCoding {
 
 		[Export ("initWithItems:header:sectionIndexTitle:")]
-		NativeHandle Constructor (CPListItem [] items, [NullAllowed] string header, [NullAllowed] string sectionIndexTitle);
+		NativeHandle Constructor (ICPListTemplateItem [] items, [NullAllowed] string header, [NullAllowed] string sectionIndexTitle);
```
The changes are correct but I believe they are a breaking change which means that we have to provide a path (even when the constructor was wrong). We need to do the following:
```suggestion
#if !XAMCORE_5_0
  		[Export ("initWithItems:header:sectionIndexTitle:")]
  		[Obsolete ("Use  'CPListSection (ICPListTemplateItem [], string, string)' constructor instead.")]
  		NativeHandle Constructor (CPListItem [] items, [NullAllowed] string header, [NullAllowed] string sectionIndexTitle);
#endif
		[Export ("initWithItems:header:sectionIndexTitle:")]
		NativeHandle Constructor (ICPListTemplateItem [] items, [NullAllowed] string header, [NullAllowed] string sectionIndexTitle);
```

mandel-macaque replied:
@dalexsoto  there was a way to allow to methods to have the same selector, right? I don't want to make this change more complicated that it is.

mandel-macaque commented on src/carplay.cs:
```diff
@@ -573,10 +573,10 @@ interface CPListItem : CPSelectableListItem, NSSecureCoding {
 
 		[Export ("initWithItems:")]
-		NativeHandle Constructor (CPListItem [] items);
+		NativeHandle Constructor (ICPListTemplateItem [] items);
```
Same as the above comment.

mandel-macaque reviewed (changes requested):
I believe we need to Obsolete and leave the old ones while providing a path for users to upgrade rather that break the API. Nevertheless I have triggered the build to confirm the API will be broken.

mandel-macaque commented:
@janwiebe-jump Thanks for the fix! I have some small concerts which I left in the review.

janwiebe-jump commented:
Thanks @mandel-macaque , good catch indeed. 
I re-added the old constructors, and added the Obsolete attribute.
Next to that, I also added the `Items[]` property (besides the existing) to conform to the iOS api.

mandel-macaque commented on src/carplay.cs:
```diff
@@ -573,11 +573,19 @@ interface CPListItem : CPSelectableListItem, NSSecureCoding {
 	interface CPListSection : NSSecureCoding {
 
 		[Export ("initWithItems:header:sectionIndexTitle:")]
+		[Obsolete ("Use 'CPListSection (ICPListTemplateItem [], string, string)' constructor instead.")]
 		NativeHandle Constructor (CPListItem [] items, [NullAllowed] string header, [NullAllowed] string sectionIndexTitle);
```
Lets wrap the wrong API in !XAMCORE_5_0 so that we can remove it the next time we have breaking changes.
```suggestion
#if !XAMCORE_5_0
		[Export ("initWithItems:header:sectionIndexTitle:")]
		[Obsolete ("Use 'CPListSection (ICPListTemplateItem [], string, string)' constructor instead.")]
		NativeHandle Constructor (CPListItem [] items, [NullAllowed] string header, [NullAllowed] string sectionIndexTitle);
#endif
```

mandel-macaque commented on src/carplay.cs:
```diff
@@ -573,11 +573,19 @@ interface CPListItem : CPSelectableListItem, NSSecureCoding {
+		[Obsolete ("Use 'CPListSection (ICPListTemplateItem [], string, string)' constructor instead.")]
 		NativeHandle Constructor (CPListItem [] items, [NullAllowed] string header, [NullAllowed] string sectionIndexTitle);
 
 		[Export ("initWithItems:")]
+		[Obsolete ("Use 'CPListSection (ICPListTemplateItem [])' constructor instead.")]
 		NativeHandle Constructor (CPListItem [] items);
```
Same:
```suggestion
#if !XAMCORE_5_0
		[Export ("initWithItems:")]
		[Obsolete ("Use 'CPListSection (ICPListTemplateItem [])' constructor instead.")]
		NativeHandle Constructor (CPListItem [] items);
#endif
```

mandel-macaque commented on src/carplay.cs:
```diff
@@ -589,8 +597,12 @@ interface CPListSection : NSSecureCoding {
 		string SectionIndexTitle { get; }
 
 		[Export ("items", ArgumentSemantic.Copy)]
+		[Obsolete ("Use 'Items : ICPListTemplateItem []' instead.")]
 		CPListItem [] Items { get; }
```
Same:
```suggestion
#if !XAMCORE_5_0
		[Export ("items", ArgumentSemantic.Copy)]
		[Obsolete ("Use 'Items : ICPListTemplateItem []' instead.")]
		CPListItem [] Items { get; }
#endif
```

mandel-macaque reviewed (changes requested):
We still need to wrap the old api with a #if so that next time we have the chance to break APIs we do not expose something wrong :)

mandel-macaque reviewed (approved):
Lets see what the tests got to see. But the changes look good to me.

mandel-macaque commented:
/azp run

janwiebe-jump commented:
Hmm, the `Items` property

> Lets see what the tests got to see. But the changes look good to me.

Thanks. I am curious how the duplicate `Items[]` works out

mandel-macaque commented:
@janwiebe-jump I suspect is going to fail, but it an easy fix (ugly, but doable). I'll keep an eye as according tot he failures I might propose a code change.

vs-mobiletools-engineering-service2 commented:
# :fire: [PR Build] Build failed :fire:

Build failed for the job 'Detect API changes'

[Pipeline](https://dev.azure.com/devdiv/DevDiv/_build/index?buildId=6520618&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [](https://github.com/xamarin/xamarin-macios/pull/15642/commits/)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

vs-mobiletools-engineering-service2 commented:
:fire: Unable to find the contents for the comment: D:\a\1\s\change-detection\results\gh-comment.md does not exist :fire

[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=6520618&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [671786172294a85a483a52739a29f95e8292f365](https://github.com/xamarin/xamarin-macios/pull/15642/commits/671786172294a85a483a52739a29f95e8292f365)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

mandel-macaque commented on src/carplay.cs:
```diff
@@ -588,8 +600,14 @@ interface CPListSection : NSSecureCoding {
 		[NullAllowed, Export ("sectionIndexTitle")]
 		string SectionIndexTitle { get; }
 
+#if !XAMCORE_5_0
 		[Export ("items", ArgumentSemantic.Copy)]
+		[Obsolete ("Use 'Items : ICPListTemplateItem []' instead.")]
 		CPListItem [] Items { get; }
+#endif
+
+		[Export ("items", ArgumentSemantic.Copy)]
+		ICPListTemplateItem [] Items { get; }
```
As I expected, having two properties with the same name results in an error:
```
carplay.cs(610,26): error CS0102: The type 'CPListSection' already contains a definition for 'Items'
```
The following should fix the issue:
```suggestion
#if !XAMCORE_5_0
		[Sealed]
		[Export ("items", ArgumentSemantic.Copy)]
		[Obsolete ("Use 'Items2 : ICPListTemplateItem []' instead.")]
		CPListItem [] Items { get; }
#eendif

		[Export ("items", ArgumentSemantic.Copy)]
#if !XAMCORE_5_0
		ICPListTemplateItem [] Items2 { get; }
#else
		ICPListTemplateItem [] Items { get; }
#endif
```
That means we have Items2 but in the future it will be Items.

dalexsoto commented on src/carplay.cs:
```diff
@@ -572,11 +572,23 @@ interface CPListItem : CPSelectableListItem, NSSecureCoding {
 
+#if !XAMCORE_5_0
 		[Export ("initWithItems:header:sectionIndexTitle:")]
+		[Obsolete ("Use 'CPListSection (ICPListTemplateItem [], string, string)' constructor instead.")]
```
```suggestion
		[Obsolete ("Use '.ctor (ICPListTemplateItem [], string, string)' constructor instead.")]
```

dalexsoto commented on src/carplay.cs:
```diff
@@ -572,11 +572,23 @@ interface CPListItem : CPSelectableListItem, NSSecureCoding {
 
+#if !XAMCORE_5_0
 		[Export ("initWithItems:")]
+		[Obsolete ("Use 'CPListSection (ICPListTemplateItem [])' constructor instead.")]
```
```suggestion
		[Obsolete ("Use '.ctor (ICPListTemplateItem [])' constructor instead.")]
```

dalexsoto commented on src/carplay.cs:
```diff
@@ -588,8 +600,19 @@ interface CPListSection : NSSecureCoding {
 		[Export ("items", ArgumentSemantic.Copy)]
+		[Obsolete ("Use 'Items2 : ICPListTemplateItem []' instead.")]
 		CPListItem [] Items { get; }
+#eendif
```
```suggestion
#endif
```

mandel-macaque commented:
/azp run

vs-mobiletools-engineering-service2 commented:
# :fire: [PR Build] Build failed :fire:

Build failed for the job 'Build packages'

[Pipeline](https://dev.azure.com/devdiv/DevDiv/_build/index?buildId=6522427&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [](https://github.com/xamarin/xamarin-macios/pull/15642/commits/)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

vs-mobiletools-engineering-service2 commented:
# :fire: [PR Build] Build failed :fire:

Build failed for the job 'Detect API changes'

[Pipeline](https://dev.azure.com/devdiv/DevDiv/_build/index?buildId=6522427&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [](https://github.com/xamarin/xamarin-macios/pull/15642/commits/)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

vs-mobiletools-engineering-service2 commented:
:fire: Unable to find the contents for the comment: D:\a\1\s\change-detection\results\gh-comment.md does not exist :fire

[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=6522427&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [93f9fb8f208d9b24cd539bbb56f6f77779d3b7fc](https://github.com/xamarin/xamarin-macios/pull/15642/commits/93f9fb8f208d9b24cd539bbb56f6f77779d3b7fc)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

janwiebe-jump commented:
The build is failing, is there anything I can do to fix it?

tj-devel709 commented:
@janwiebe-jump 
Seems that the build is failing due to:
```
/Library/Frameworks/Xamarin.iOS.framework/Versions/15.13.0.27/src/Xamarin.iOS/build/dotnet/ios/generated-sources/CarPlay/CPListSection.g.cs(157): error MT4119: Could not register the selector 'initWithItems:header:sectionIndexTitle:' of the member 'CarPlay.CPListSection..ctor' because the selector is already registered on the member '.ctor'.
		
/Library/Frameworks/Xamarin.iOS.framework/Versions/15.13.0.27/src/Xamarin.iOS/build/dotnet/ios/generated-sources/CarPlay/CPListSection.g.cs(184): error MT4119: Could not register the selector 'initWithItems:' of the member 'CarPlay.CPListSection..ctor' because the selector is already registered on the member '.ctor'.
```

chamons commented:
I suspect that error is due to double exporting the ctor selectors. Let me take a look.

chamons commented:
I believe I have corrected the PR. It at least compiles for me. I will check tomorrow.

chamons commented:
/azp run

vs-mobiletools-engineering-service2 commented:
# :computer: [PR Build] Tests on macOS Mac Catalina (10.15) passed :computer:

:white_check_mark: **All** tests on macOS Mac Catalina (10.15) passed.

[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=6533344&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [4ae2609c3901a3b0e5fa8e1fd93a32260c750487](https://github.com/xamarin/xamarin-macios/pull/15642/commits/4ae2609c3901a3b0e5fa8e1fd93a32260c750487)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

vs-mobiletools-engineering-service2 commented:
# :books: [PR Build] Artifacts :books:

## Packages generated

<details><summary>View packages</summary>

* [Microsoft.iOS.Bundle.15.4.337-ci.pr.gh15642.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.iOS.Bundle.15.4.337-ci.pr.gh15642.pkg)
* [Microsoft.iOS.Ref.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.iOS.Ref.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.iOS.Runtime.ios-arm.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.iOS.Runtime.ios-arm.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.iOS.Runtime.ios-arm64.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.iOS.Runtime.ios-arm64.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.iOS.Runtime.iossimulator-arm64.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.iOS.Runtime.iossimulator-arm64.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.iOS.Runtime.iossimulator-x64.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.iOS.Runtime.iossimulator-x64.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.iOS.Runtime.iossimulator-x86.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.iOS.Runtime.iossimulator-x86.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.iOS.Sdk.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.iOS.Sdk.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.iOS.Templates.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.iOS.Templates.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.iOS.Windows.Sdk.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.iOS.Windows.Sdk.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.MacCatalyst.Bundle.15.4.337-ci.pr.gh15642.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.MacCatalyst.Bundle.15.4.337-ci.pr.gh15642.pkg)
* [Microsoft.MacCatalyst.Ref.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.MacCatalyst.Ref.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.MacCatalyst.Runtime.maccatalyst-arm64.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.MacCatalyst.Runtime.maccatalyst-arm64.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.MacCatalyst.Runtime.maccatalyst-x64.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.MacCatalyst.Runtime.maccatalyst-x64.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.MacCatalyst.Sdk.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.MacCatalyst.Sdk.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.MacCatalyst.Templates.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.MacCatalyst.Templates.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.macOS.Bundle.12.3.337-ci.pr.gh15642.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.macOS.Bundle.12.3.337-ci.pr.gh15642.pkg)
* [Microsoft.macOS.Ref.12.3.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.macOS.Ref.12.3.337-ci.pr.gh15642.nupkg)
* [Microsoft.macOS.Runtime.osx-arm64.12.3.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.macOS.Runtime.osx-arm64.12.3.337-ci.pr.gh15642.nupkg)
* [Microsoft.macOS.Runtime.osx-x64.12.3.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.macOS.Runtime.osx-x64.12.3.337-ci.pr.gh15642.nupkg)
* [Microsoft.macOS.Sdk.12.3.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.macOS.Sdk.12.3.337-ci.pr.gh15642.nupkg)
* [Microsoft.macOS.Templates.12.3.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.macOS.Templates.12.3.337-ci.pr.gh15642.nupkg)
* [Microsoft.NET.Sdk.iOS.Manifest-6.0.300.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.NET.Sdk.iOS.Manifest-6.0.300.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.NET.Sdk.MacCatalyst.Manifest-6.0.300.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.NET.Sdk.MacCatalyst.Manifest-6.0.300.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.NET.Sdk.macOS.Manifest-6.0.300.12.3.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.NET.Sdk.macOS.Manifest-6.0.300.12.3.337-ci.pr.gh15642.nupkg)
* [Microsoft.NET.Sdk.tvOS.Manifest-6.0.300.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.NET.Sdk.tvOS.Manifest-6.0.300.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.NET.Workload.iOS.15.4.337-ci.pr.gh15642.msi](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.NET.Workload.iOS.15.4.337-ci.pr.gh15642.msi)
* [Microsoft.tvOS.Bundle.15.4.337-ci.pr.gh15642.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.tvOS.Bundle.15.4.337-ci.pr.gh15642.pkg)
* [Microsoft.tvOS.Ref.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.tvOS.Ref.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.tvOS.Runtime.tvos-arm64.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.tvOS.Runtime.tvos-arm64.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.tvOS.Runtime.tvossimulator-arm64.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.tvOS.Runtime.tvossimulator-arm64.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.tvOS.Runtime.tvossimulator-x64.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.tvOS.Runtime.tvossimulator-x64.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.tvOS.Sdk.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.tvOS.Sdk.15.4.337-ci.pr.gh15642.nupkg)
* [Microsoft.tvOS.Templates.15.4.337-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/Microsoft.tvOS.Templates.15.4.337-ci.pr.gh15642.nupkg)
* [xamarin.ios-15.13.0.33.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/xamarin.ios-15.13.0.33.pkg)
* [xamarin.mac-8.13.0.33.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/be64ee580d47e288a93cfdfc74fe23d27debb07a/6533344/package/xamarin.mac-8.13.0.33.pkg)
</details>

[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=6533344&view=ms.vss-test-web.test-result-details) on Agent XAMBOT-1042.Monterey'
Hash: [4ae2609c3901a3b0e5fa8e1fd93a32260c750487](https://github.com/xamarin/xamarin-macios/pull/15642/commits/4ae2609c3901a3b0e5fa8e1fd93a32260c750487)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

vs-mobiletools-engineering-service2 commented:
## :white_check_mark: API diff for current PR / commit

<details><summary>Legacy Xamarin (No breaking changes)</summary>

* ~iOS~ (no change detected)
* ~tvOS~ (no change detected)
* ~watchOS~ (no change detected)
* ~macOS~ (no change detected)

</details>

<details><summary>NET (empty diffs)</summary>

* ~iOS~: (empty diff detected)
* ~tvOS~: (empty diff detected)
* ~MacCatalyst~: (empty diff detected)
* ~macOS~: (empty diff detected)

</details>


## :heavy_exclamation_mark: API diff vs stable (Breaking changes)

<details><summary>Legacy Xamarin (:heavy_exclamation_mark: Breaking changes :heavy_exclamation_mark:)</summary>

* iOS: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/detected-changes/20220809.3/6533344/;/stable-api-comparison/ios-api-diff.html) [gist](https://gist.github.com/f307a6a533482d068a618fc5de2626f2) (:heavy_exclamation_mark: Breaking changes :heavy_exclamation_mark:)
* tvOS: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/detected-changes/20220809.3/6533344/;/stable-api-comparison/tvos-api-diff.html) [gist](https://gist.github.com/3dedf1edcf6a32f7d06e62e9b3cfea09) (No breaking changes)
* watchOS: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/detected-changes/20220809.3/6533344/;/stable-api-comparison/watchos-api-diff.html) [gist](https://gist.github.com/d5cfb7c2905796e8de3fc520ff88b982) (No breaking changes)
* macOS: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/detected-changes/20220809.3/6533344/;/stable-api-comparison/mac-api-diff.html) [gist](https://gist.github.com/0b3a1980bf4492bf8ac600a25eed4c2c) (No breaking changes)

</details>

<details><summary>.NET (:heavy_exclamation_mark: Breaking changes :heavy_exclamation_mark:)</summary>

* iOS: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/detected-changes/20220809.3/6533344/;/stable-api-comparison/diff/dotnet/Microsoft.iOS.Ref/ref/net6.0/Microsoft.iOS.html) [gist](https://gist.github.com/3aba86ee1389ca834e56af08c21ab5a2) (:heavy_exclamation_mark: Breaking changes :heavy_exclamation_mark:)
* tvOS: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/detected-changes/20220809.3/6533344/;/stable-api-comparison/diff/dotnet/Microsoft.tvOS.Ref/ref/net6.0/Microsoft.tvOS.html) [gist](https://gist.github.com/a529b2e8061520fd87db34454fcb2d8e) (No breaking changes)
* MacCatalyst: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/detected-changes/20220809.3/6533344/;/stable-api-comparison/diff/dotnet/Microsoft.MacCatalyst.Ref/ref/net6.0/Microsoft.MacCatalyst.html) [gist](https://gist.github.com/f551d25d88b85dd077095568caee7067) (No breaking changes)
* macOS: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/detected-changes/20220809.3/6533344/;/stable-api-comparison/diff/dotnet/Microsoft.macOS.Ref/ref/net6.0/Microsoft.macOS.html) [gist](https://gist.github.com/bb94e8cecbd7063ad4eab9d9f5d146f6) (No breaking changes)
* Microsoft.iOS vs Microsoft.MacCatalyst: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/detected-changes/20220809.3/6533344/;/stable-api-comparison/diff/dotnet/iOS-MacCatalyst-diff/Microsoft.iOS.Ref/ref/net6.0/Microsoft.iOS.MacCatalyst.html) [gist](https://gist.github.com/167ae9c5b9b60dc6e9c29100de363a8f)

</details>

<details><summary>Legacy Xamarin (stable) vs .NET</summary>

* iOS: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/detected-changes/20220809.3/6533344/;/stable-api-comparison/diff/dotnet/legacy-diff/Microsoft.iOS.Ref/ref/net6.0/Microsoft.iOS.html) [gist](https://gist.github.com/bac642c8b9ad676f9990ebd08fae4275)
* tvOS: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/detected-changes/20220809.3/6533344/;/stable-api-comparison/diff/dotnet/legacy-diff/Microsoft.tvOS.Ref/ref/net6.0/Microsoft.tvOS.html) [gist](https://gist.github.com/e43697ab862556e807d81e40ad51c179)
* macOS: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/detected-changes/20220809.3/6533344/;/stable-api-comparison/diff/dotnet/legacy-diff/Microsoft.macOS.Ref/ref/net6.0/Microsoft.macOS.html) [gist](https://gist.github.com/19da1fea4dc5b84e30bbb2554081e5d8)

</details>


<!-- BreakingChangesDetected -->


## :information_source: Generator diff

Generator Diff: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/detected-changes/20220809.3/6533344/;/generator-diff/index.html) (html) [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/detected-changes/20220809.3/6533344/;/generator-diff/generator.diff) (raw diff) Unable to create gist: Response status code does not indicate success: 502 (Bad Gateway). (raw diff) - Please review changes)


[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=6533344&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [4ae2609c3901a3b0e5fa8e1fd93a32260c750487](https://github.com/xamarin/xamarin-macios/pull/15642/commits/4ae2609c3901a3b0e5fa8e1fd93a32260c750487)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

vs-mobiletools-engineering-service2 commented:
# :x: [PR Build] Tests on macOS M1 - Mac Big Sur (11.5) failed :x:

Failed tests are:

* xammac_tests
* monotouch-test


[Pipeline](https://dev.azure.com/devdiv/DevDiv/_build/index?buildId=6533344&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [4ae2609c3901a3b0e5fa8e1fd93a32260c750487](https://github.com/xamarin/xamarin-macios/pull/15642/commits/4ae2609c3901a3b0e5fa8e1fd93a32260c750487)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

vs-mobiletools-engineering-service2 commented:
# :fire: [CI Build] Test results :fire:

# Test results
:x: Tests failed on VSTS: simulator tests

:tada: All 221 tests passed :tada:

## Failures

### :x: msbuild tests

:fire: Failed catastrophically on VSTS: simulator tests - msbuild (no summary found).

[Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220809.3/6533344//simulator_msbuild/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6533344/artifacts?artifactName=HtmlReport-simulator_msbuild&api-version=6.0&$format=zip)

## Successes

:white_check_mark: bcl: All 69 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220809.3/6533344//simulator_bcl/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6533344/artifacts?artifactName=HtmlReport-simulator_bcl&api-version=6.0&$format=zip)
:white_check_mark: cecil: All 1 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220809.3/6533344//simulator_cecil/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6533344/artifacts?artifactName=HtmlReport-simulator_cecil&api-version=6.0&$format=zip)
:white_check_mark: dotnettests: All 1 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220809.3/6533344//simulator_dotnettests/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6533344/artifacts?artifactName=HtmlReport-simulator_dotnettests&api-version=6.0&$format=zip)
:white_check_mark: fsharp: All 7 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220809.3/6533344//simulator_fsharp/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6533344/artifacts?artifactName=HtmlReport-simulator_fsharp&api-version=6.0&$format=zip)
:white_check_mark: framework: All 8 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220809.3/6533344//simulator_framework/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6533344/artifacts?artifactName=HtmlReport-simulator_framework&api-version=6.0&$format=zip)
:white_check_mark: generator: All 2 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220809.3/6533344//simulator_generator/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6533344/artifacts?artifactName=HtmlReport-simulator_generator&api-version=6.0&$format=zip)
:white_check_mark: interdependent_binding_projects: All 7 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220809.3/6533344//simulator_interdependent_binding_projects/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6533344/artifacts?artifactName=HtmlReport-simulator_interdependent_binding_projects&api-version=6.0&$format=zip)
:white_check_mark: install_source: All 1 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220809.3/6533344//simulator_install_source/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6533344/artifacts?artifactName=HtmlReport-simulator_install_source&api-version=6.0&$format=zip)
:white_check_mark: introspection: All 8 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220809.3/6533344//simulator_introspection/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6533344/artifacts?artifactName=HtmlReport-simulator_introspection&api-version=6.0&$format=zip)
:white_check_mark: linker: All 65 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220809.3/6533344//simulator_linker/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6533344/artifacts?artifactName=HtmlReport-simulator_linker&api-version=6.0&$format=zip)
:white_check_mark: mac_binding_project: All 1 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220809.3/6533344//simulator_mac_binding_project/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6533344/artifacts?artifactName=HtmlReport-simulator_mac_binding_project&api-version=6.0&$format=zip)
:white_check_mark: mmp: All 2 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220809.3/6533344//simulator_mmp/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6533344/artifacts?artifactName=HtmlReport-simulator_mmp&api-version=6.0&$format=zip)
:white_check_mark: mononative: All 12 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220809.3/6533344//simulator_mononative/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6533344/artifacts?artifactName=HtmlReport-simulator_mononative&api-version=6.0&$format=zip)
:white_check_mark: monotouch: All 23 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220809.3/6533344//simulator_monotouch/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6533344/artifacts?artifactName=HtmlReport-simulator_monotouch&api-version=6.0&$format=zip)
:white_check_mark: mtouch: All 1 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220809.3/6533344//simulator_mtouch/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6533344/artifacts?artifactName=HtmlReport-simulator_mtouch&api-version=6.0&$format=zip)
:white_check_mark: xammac: All 3 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220809.3/6533344//simulator_xammac/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6533344/artifacts?artifactName=HtmlReport-simulator_xammac&api-version=6.0&$format=zip)
:white_check_mark: xcframework: All 8 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220809.3/6533344//simulator_xcframework/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6533344/artifacts?artifactName=HtmlReport-simulator_xcframework&api-version=6.0&$format=zip)
:white_check_mark: xtro: All 2 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220809.3/6533344//simulator_xtro/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6533344/artifacts?artifactName=HtmlReport-simulator_xtro&api-version=6.0&$format=zip)

[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=6533344&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [](https://github.com/xamarin/xamarin-macios/pull/15642/commits/)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

janwiebe-jump commented:
Hi all, thanks for your reviews and input.
I would like to know what is next to get this merged and released.
It blocks the release of our app. Is there anything I can do?
Thanks!

dalexsoto commented:
/azp run

vs-mobiletools-engineering-service2 commented:
# :fire: [PR Build] Build failed :fire:

Build failed for the job 'Build packages'

[Pipeline](https://dev.azure.com/devdiv/DevDiv/_build/index?buildId=6551437&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [](https://github.com/xamarin/xamarin-macios/pull/15642/commits/)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

vs-mobiletools-engineering-service2 commented:
# :fire: [PR Build] Build failed :fire:

Build failed for the job 'Detect API changes'

[Pipeline](https://dev.azure.com/devdiv/DevDiv/_build/index?buildId=6551437&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [](https://github.com/xamarin/xamarin-macios/pull/15642/commits/)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

dalexsoto commented:
Looks like our CI is having a hard time :(

vs-mobiletools-engineering-service2 commented:
# :fire: [PR Build] Build failed :fire:

Build failed for the job 'Build packages'

[Pipeline](https://dev.azure.com/devdiv/DevDiv/_build/index?buildId=6558944&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [](https://github.com/xamarin/xamarin-macios/pull/15642/commits/)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

vs-mobiletools-engineering-service2 commented:
:fire: Unable to find the contents for the comment: D:\a\1\s\change-detection\results\gh-comment.md does not exist :fire

[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=6558944&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [8e48b2dbfebfc6aacf048270c31d6cedd17f4640](https://github.com/xamarin/xamarin-macios/pull/15642/commits/8e48b2dbfebfc6aacf048270c31d6cedd17f4640)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

rolfbjarne commented:
/azp run

vs-mobiletools-engineering-service2 commented:
# :fire: [PR Build] Build failed :fire:

Build failed for the job 'Build packages'

[Pipeline](https://dev.azure.com/devdiv/DevDiv/_build/index?buildId=6576502&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [](https://github.com/xamarin/xamarin-macios/pull/15642/commits/)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

vs-mobiletools-engineering-service2 commented:
# :fire: [PR Build] Build failed :fire:

Build failed for the job 'Detect API changes'

[Pipeline](https://dev.azure.com/devdiv/DevDiv/_build/index?buildId=6576502&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [](https://github.com/xamarin/xamarin-macios/pull/15642/commits/)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

vs-mobiletools-engineering-service2 commented:
:fire: Unable to find the contents for the comment: D:\a\1\s\change-detection\results\gh-comment.md does not exist :fire

[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=6576502&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [d7adeb5220a82c35bfdad32a553f6354d06d5e7a](https://github.com/xamarin/xamarin-macios/pull/15642/commits/d7adeb5220a82c35bfdad32a553f6354d06d5e7a)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

rolfbjarne commented:
It doesn't compile:

```
build/dotnet/ios/generated-sources/CarPlay/CPListSection.g.cs(387,42): error CS8115: A throw expression is not allowed in this context.
build/dotnet/ios/generated-sources/CarPlay/CPListSection.g.cs(387,48): error CS0039: Cannot convert type 'System.InvalidOperationException' to 'Foundation.NSArray' via a reference conversion, boxing conversion, unboxing conversion, wrapping conversion, or null type conversion
make[1]: *** [build/dotnet/ios/32/Microsoft.iOS.dll] Error 1
```

dalexsoto commented:
/azp run

vs-mobiletools-engineering-service2 commented:
:fire: Failed to compare API and create generator diff :fire:

Error: 'make' failed for the hash 1f1227071315bf6fa6c94ed907fe37af26ff408e.


[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=6586764&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [1fcaeca530e72a118d9e4292b849dd9aedc199bf](https://github.com/xamarin/xamarin-macios/pull/15642/commits/1fcaeca530e72a118d9e4292b849dd9aedc199bf)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

vs-mobiletools-engineering-service2 commented:
# :books: [PR Build] Artifacts :books:

## Packages generated

<details><summary>View packages</summary>

* [Microsoft.iOS.Bundle.15.4.349-ci.pr.gh15642.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.iOS.Bundle.15.4.349-ci.pr.gh15642.pkg)
* [Microsoft.iOS.Ref.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.iOS.Ref.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.iOS.Runtime.ios-arm.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.iOS.Runtime.ios-arm.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.iOS.Runtime.ios-arm64.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.iOS.Runtime.ios-arm64.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.iOS.Runtime.iossimulator-arm64.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.iOS.Runtime.iossimulator-arm64.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.iOS.Runtime.iossimulator-x64.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.iOS.Runtime.iossimulator-x64.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.iOS.Runtime.iossimulator-x86.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.iOS.Runtime.iossimulator-x86.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.iOS.Sdk.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.iOS.Sdk.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.iOS.Templates.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.iOS.Templates.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.iOS.Windows.Sdk.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.iOS.Windows.Sdk.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.MacCatalyst.Bundle.15.4.349-ci.pr.gh15642.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.MacCatalyst.Bundle.15.4.349-ci.pr.gh15642.pkg)
* [Microsoft.MacCatalyst.Ref.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.MacCatalyst.Ref.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.MacCatalyst.Runtime.maccatalyst-arm64.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.MacCatalyst.Runtime.maccatalyst-arm64.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.MacCatalyst.Runtime.maccatalyst-x64.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.MacCatalyst.Runtime.maccatalyst-x64.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.MacCatalyst.Sdk.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.MacCatalyst.Sdk.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.MacCatalyst.Templates.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.MacCatalyst.Templates.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.macOS.Bundle.12.3.349-ci.pr.gh15642.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.macOS.Bundle.12.3.349-ci.pr.gh15642.pkg)
* [Microsoft.macOS.Ref.12.3.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.macOS.Ref.12.3.349-ci.pr.gh15642.nupkg)
* [Microsoft.macOS.Runtime.osx-arm64.12.3.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.macOS.Runtime.osx-arm64.12.3.349-ci.pr.gh15642.nupkg)
* [Microsoft.macOS.Runtime.osx-x64.12.3.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.macOS.Runtime.osx-x64.12.3.349-ci.pr.gh15642.nupkg)
* [Microsoft.macOS.Sdk.12.3.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.macOS.Sdk.12.3.349-ci.pr.gh15642.nupkg)
* [Microsoft.macOS.Templates.12.3.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.macOS.Templates.12.3.349-ci.pr.gh15642.nupkg)
* [Microsoft.NET.Sdk.iOS.Manifest-6.0.300.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.NET.Sdk.iOS.Manifest-6.0.300.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.NET.Sdk.MacCatalyst.Manifest-6.0.300.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.NET.Sdk.MacCatalyst.Manifest-6.0.300.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.NET.Sdk.macOS.Manifest-6.0.300.12.3.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.NET.Sdk.macOS.Manifest-6.0.300.12.3.349-ci.pr.gh15642.nupkg)
* [Microsoft.NET.Sdk.tvOS.Manifest-6.0.300.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.NET.Sdk.tvOS.Manifest-6.0.300.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.NET.Workload.iOS.15.4.349-ci.pr.gh15642.msi](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.NET.Workload.iOS.15.4.349-ci.pr.gh15642.msi)
* [Microsoft.tvOS.Bundle.15.4.349-ci.pr.gh15642.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.tvOS.Bundle.15.4.349-ci.pr.gh15642.pkg)
* [Microsoft.tvOS.Ref.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.tvOS.Ref.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.tvOS.Runtime.tvos-arm64.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.tvOS.Runtime.tvos-arm64.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.tvOS.Runtime.tvossimulator-arm64.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.tvOS.Runtime.tvossimulator-arm64.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.tvOS.Runtime.tvossimulator-x64.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.tvOS.Runtime.tvossimulator-x64.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.tvOS.Sdk.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.tvOS.Sdk.15.4.349-ci.pr.gh15642.nupkg)
* [Microsoft.tvOS.Templates.15.4.349-ci.pr.gh15642.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/Microsoft.tvOS.Templates.15.4.349-ci.pr.gh15642.nupkg)
* [xamarin.ios-15.13.0.45.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/xamarin.ios-15.13.0.45.pkg)
* [xamarin.mac-8.13.0.45.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/merge/86b68eb0a0904802d58d39199f3502a1e820d398/6586764/package/xamarin.mac-8.13.0.45.pkg)
</details>

[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=6586764&view=ms.vss-test-web.test-result-details) on Agent XAMBOT-1109.Monterey'
Hash: [1fcaeca530e72a118d9e4292b849dd9aedc199bf](https://github.com/xamarin/xamarin-macios/pull/15642/commits/1fcaeca530e72a118d9e4292b849dd9aedc199bf)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

vs-mobiletools-engineering-service2 commented:
# :computer: [PR Build] Tests on macOS Mac Catalina (10.15) passed :computer:

:white_check_mark: **All** tests on macOS Mac Catalina (10.15) passed.

[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=6586764&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [1fcaeca530e72a118d9e4292b849dd9aedc199bf](https://github.com/xamarin/xamarin-macios/pull/15642/commits/1fcaeca530e72a118d9e4292b849dd9aedc199bf)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

vs-mobiletools-engineering-service2 commented:
# :x: [PR Build] Tests on macOS M1 - Mac Big Sur (11.5) failed :x:

Failed tests are:

* introspection
* xammac_tests
* monotouch-test


[Pipeline](https://dev.azure.com/devdiv/DevDiv/_build/index?buildId=6586764&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [1fcaeca530e72a118d9e4292b849dd9aedc199bf](https://github.com/xamarin/xamarin-macios/pull/15642/commits/1fcaeca530e72a118d9e4292b849dd9aedc199bf)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

vs-mobiletools-engineering-service2 commented:
# :rocket: [CI Build] Test results :rocket:

# Test results
:white_check_mark: All tests passed on VSTS: simulator tests.

:tada: All 223 tests passed :tada:

## Tests counts
:white_check_mark: bcl: All 69 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220820.2/6586764//simulator_bcl/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6586764/artifacts?artifactName=HtmlReport-simulator_bcl&api-version=6.0&$format=zip)
:white_check_mark: cecil: All 1 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220820.2/6586764//simulator_cecil/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6586764/artifacts?artifactName=HtmlReport-simulator_cecil&api-version=6.0&$format=zip)
:white_check_mark: dotnettests: All 1 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220820.2/6586764//simulator_dotnettests/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6586764/artifacts?artifactName=HtmlReport-simulator_dotnettests&api-version=6.0&$format=zip)
:white_check_mark: fsharp: All 7 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220820.2/6586764//simulator_fsharp/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6586764/artifacts?artifactName=HtmlReport-simulator_fsharp&api-version=6.0&$format=zip)
:white_check_mark: framework: All 8 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220820.2/6586764//simulator_framework/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6586764/artifacts?artifactName=HtmlReport-simulator_framework&api-version=6.0&$format=zip)
:white_check_mark: generator: All 2 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220820.2/6586764//simulator_generator/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6586764/artifacts?artifactName=HtmlReport-simulator_generator&api-version=6.0&$format=zip)
:white_check_mark: interdependent_binding_projects: All 7 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220820.2/6586764//simulator_interdependent_binding_projects/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6586764/artifacts?artifactName=HtmlReport-simulator_interdependent_binding_projects&api-version=6.0&$format=zip)
:white_check_mark: install_source: All 1 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220820.2/6586764//simulator_install_source/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6586764/artifacts?artifactName=HtmlReport-simulator_install_source&api-version=6.0&$format=zip)
:white_check_mark: introspection: All 8 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220820.2/6586764//simulator_introspection/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6586764/artifacts?artifactName=HtmlReport-simulator_introspection&api-version=6.0&$format=zip)
:white_check_mark: linker: All 65 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220820.2/6586764//simulator_linker/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6586764/artifacts?artifactName=HtmlReport-simulator_linker&api-version=6.0&$format=zip)
:white_check_mark: mac_binding_project: All 1 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220820.2/6586764//simulator_mac_binding_project/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6586764/artifacts?artifactName=HtmlReport-simulator_mac_binding_project&api-version=6.0&$format=zip)
:white_check_mark: mmp: All 2 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220820.2/6586764//simulator_mmp/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6586764/artifacts?artifactName=HtmlReport-simulator_mmp&api-version=6.0&$format=zip)
:white_check_mark: mononative: All 12 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220820.2/6586764//simulator_mononative/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6586764/artifacts?artifactName=HtmlReport-simulator_mononative&api-version=6.0&$format=zip)
:white_check_mark: monotouch: All 23 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220820.2/6586764//simulator_monotouch/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6586764/artifacts?artifactName=HtmlReport-simulator_monotouch&api-version=6.0&$format=zip)
:white_check_mark: msbuild: All 2 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220820.2/6586764//simulator_msbuild/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6586764/artifacts?artifactName=HtmlReport-simulator_msbuild&api-version=6.0&$format=zip)
:white_check_mark: mtouch: All 1 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220820.2/6586764//simulator_mtouch/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6586764/artifacts?artifactName=HtmlReport-simulator_mtouch&api-version=6.0&$format=zip)
:white_check_mark: xammac: All 3 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220820.2/6586764//simulator_xammac/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6586764/artifacts?artifactName=HtmlReport-simulator_xammac&api-version=6.0&$format=zip)
:white_check_mark: xcframework: All 8 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220820.2/6586764//simulator_xcframework/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6586764/artifacts?artifactName=HtmlReport-simulator_xcframework&api-version=6.0&$format=zip)
:white_check_mark: xtro: All 2 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20220820.2/6586764//simulator_xtro/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/6586764/artifacts?artifactName=HtmlReport-simulator_xtro&api-version=6.0&$format=zip)

[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=6586764&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [](https://github.com/xamarin/xamarin-macios/pull/15642/commits/)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps)

rolfbjarne commented:
Thanks a lot for your help!

janwiebe-jump commented:
Hi @rolfbjarne,
Thanks a lot, also for your ant the team's assistance!
Will this also be released as a xamarin-ios version that I can use in my Xamarin iOS (Forms) app? We can't yet move to MAUI / .NET 6, due to a lot of dependencies, so I would strongly prefer a xamarin-ios.pkg version with this fix included, so we can release our CarPlay app.
Thanks so much!

rolfbjarne commented:
@janwiebe-jump yes, this will be released in a xamarin-ios.pkg (when we release support for Xcode 14, the packages will contain your fix)
